crypto/ec/ecdh_ossl.c

```diff
@@ -66,6 +66,10 @@ int ecdh_simple_compute_key(unsigned char **pout, size_t *poutlen,
     BN_CTX_start(ctx);
     x = BN_CTX_get(ctx);
     y = BN_CTX_get(ctx);
+    if (y == NULL) {
```

Comment:
I wasn't clear.  Looking at the original diff, suppose "thread 2" before line 66 allocates all memory.  Then before line 67 it releases it.

Please check ALL returned values.  Anythign else is confusing.  For example, this whole thread would have been avoided at the cost of an extra "jump if nonzero" assembler instrution. :)